### ISSUE TITLE
Installation directory confusion

### ISSUE THREAD

**Issue by mattst88** [opendnssec/SoftHSMv2#582]
By default the shared object libsofthsm2.so is installed to

    /usr/${libdir}/softhsm/libsofthsm2.so

[SSSD](https://github.com/SSSD/sssd) doesn't look in this location and instead looks in three others. See https://github.com/SSSD/sssd/blob/master/src/external/test_ca.m4#L36

I patched SSSD in Gentoo (https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-auth/sssd/files/sssd-2.3.1-test_ca-Look-for-libsofthsm2.so-in-usr-libdir-sofths.patch) which seems reasonable, but I noticed that Fedora modifies the SoftHSMv2 build itself. See https://src.fedoraproject.org/rpms/softhsm/blob/master/f/softhsm.spec#_46

Where should `libsofthsm2.so` be installed? Should SSSD be updated to search this additional location? Should the default installation location of libsofthsm2.so be changed?

**Comment by rijswijk** [comment]
SoftHSM is a module rather than a shared library that applications link against at compile time. This is the reason why it is installed in its own subdirectory rather than in `$libdir`. Applications are meant to load it using `dl_open(..)` at runtime (this is the normal use pattern for applications that use PKCS #11). 

This path has been the default installation path ever since SoftHSM v2.0.0.

What would, perhaps, be an option is to change `configure.ac` such that the path can be overridden during the `configure` run.

**Comment by mattst88** [comment]
To close the loop, [this commit](https://github.com/SSSD/sssd/commit/a06ce2107e0bad14e7313f218db514947d8cad2f) was made to sssd that resolves the problem. It's included in v2.4.1.

````
commit a06ce2107e0bad14e7313f218db514947d8cad2f
Author:     Marco Trevisan (Treviño) <mail@3v1n0.net>
AuthorDate: Wed Dec 16 18:19:00 2020 +0100
Commit:     Pavel Březina <pbrezina@redhat.com>
CommitDate: Thu Jan 21 10:49:21 2021 +0100

    test_ca: Look for libsofthsm2 in libdir before falling back to hardcoded paths
    
    Right now building SSSD in archs different from amd64 (at least in
    debian and derivatives) won't ever get the test_CA built because
    libsofthsm2 won't be found (leading also to #5397 at times).
    
    As per this, until they won't provide a pkg-config file:
     - Prioritize looking for libsofthsm2 in configured libdir (will help
       the developer case when using custom prefixes with custom softhsm2)
     - Fallback to /usr prefixes, supporting any arch (not only x86_64)
    
    Reviewed-by: Sumit Bose <sbose@redhat.com>
````
